### PR TITLE
stop ruby 2 support 

### PR DESCRIPTION
**Pull request by CharlesDelannoy**
We want to drop support of `ruby 2` as it's an old version and conflict with the last rails versions
We are also stopping the handle of rails/activerecord 6.0. 

Removing `ruby 2` makes u remove the use of `ruby2_keywords` 
Then we had to cleanup the args we are using when patching `ActiveRecord` methods

### CODE REVIEW

File: lib/safe-pg-migrations/plugins/idempotent_statements.rb [hunk 1]
```diff
@@ -64,7 +65,7 @@ def remove_foreign_key(from_table, to_table = nil, **options)
       log_message("/!\\ Foreign key '#{from_table}' -> '#{reference_name}' does not exist. Skipping statement.")
     end
 
-    ruby2_keywords def create_table(table_name, *args)
+    def create_table(table_name, *args)
       options = args.last.is_a?(Hash) ? args.last : {}
```
Comment (MouadHilali) [inline lib/safe-pg-migrations/plugins/idempotent_statements.rb:69]: why didn't you use `**options` here?

File: lib/safe-pg-migrations/plugins/statement_insurer.rb [hunk 2]
```diff
@@ -30,7 +30,7 @@ def validate_foreign_key(*, **)
       without_statement_timeout { super }
     end
 
-    ruby2_keywords def add_foreign_key(from_table, to_table, *args)
+    def add_foreign_key(from_table, to_table, *args)
       options = args.last.is_a?(Hash) ? args.last : {}
```
Comment (MouadHilali) [inline lib/safe-pg-migrations/plugins/statement_insurer.rb:34]: same here

File: lib/safe-pg-migrations/plugins/statement_insurer.rb [hunk 3]
```diff
@@ -53,7 +53,7 @@ def validate_foreign_key(*, **)
       end
     end
 
-    ruby2_keywords def add_index(table_name, column_name, *args_options)
+    def add_index(table_name, column_name, *args_options)
       options = args_options.last.is_a?(Hash) ? args_options.last : {}
```
Comment (MouadHilali) [inline lib/safe-pg-migrations/plugins/statement_insurer.rb:57]: and here

File: test/StatementInsurer/change_column_null_test.rb [hunk 4]
```diff
@@ -126,8 +126,8 @@ def skip_if_unmet_requirements!
     end
 
     def met_requirements?
-      Gem::Requirement.new('>6.1', '<7.1').satisfied_by?(Gem::Version.new(::ActiveRecord::VERSION::STRING)) &&
-        SafePgMigrations.pg_version_num >= 120_000
+      Gem::Requirement.new('>6.1').satisfied_by?(Gem::Version.new(::ActiveRecord::VERSION::STRING)) &&
+        SafePgMigrations.get_pg_version_num(ActiveRecord::Base.connection) >= 120_000
```
Comment (EQuincerot) [inline test/StatementInsurer/change_column_null_test.rb:130]: I'm not sure what should happen with 6.1 version, but maybe we could just keep:
```suggestion
      SafePgMigrations.get_pg_version_num(ActiveRecord::Base.connection) >= 120_000
```

File: lib/safe-pg-migrations/plugins/idempotent_statements.rb [hunk 5]
```diff
@@ -15,42 +14,44 @@ module IdempotentStatements
...
-                 )
+      if column_exists?(table_name, column_name, type)
+        log_message(<<~MESSAGE.squish
+          /!\\ Column '#{column_name}' already exists in '#{table_name}' with the same type (#{type}).
+          Skipping statement.
+        MESSAGE
+                   )
+      else
+        super(table_name, column_name, type, **options_without_default_value_backfill)
+      end
```
Comment (EQuincerot) [inline lib/safe-pg-migrations/plugins/idempotent_statements.rb:37]: Nitpick: I think you should keep the initial order if possible, to help the review of the PR and focus on mandatory changes:
```
options_without_default_value_backfill = options.except(:default_value_backfill)

return super(table_name, column_name, type, **options_without_default_value_backfill) unless column_exists?(table_name, column_name, type)

log_message(<<~MESSAGE.squish
          /!\\ Column '#{column_name}' already exists in '#{table_name}' with the same type (#{type}).
          Skipping statement.
        MESSAGE
                   )

File: .github/workflows/continuous-integration-workflow.yml [hunk 6]
```diff
@@ -66,10 +66,6 @@ jobs:
             gemfile: gemfiles/without_strong_migrations.gemfile
           - ruby: "3.0"
             gemfile: gemfiles/activerecord61.gemfile
-          - ruby: "3.0"
-            gemfile: gemfiles/activerecord60.gemfile
```
[the remark this answers is not in the document]
Comment (CharlesDelannoy) [reply .github/workflows/continuous-integration-workflow.yml:70]: yes we stop checking / support of rails 6.0

File: lib/safe-pg-migrations/plugins/idempotent_statements.rb (hunk 1 above)
Comment (CharlesDelannoy) [reply lib/safe-pg-migrations/plugins/idempotent_statements.rb:69]: indeed you're right :)

File: test/StatementInsurer/change_column_null_test.rb (hunk 4 above)
Comment (frederic-martin-doctolib) [reply test/StatementInsurer/change_column_null_test.rb:130]: why do you need to remove '<7.1' ?

Comment (EQuincerot) [reply test/StatementInsurer/change_column_null_test.rb:130]: 6.1, not 7.1 :)

File: lib/safe-pg-migrations/plugins/idempotent_statements.rb (hunk 5 above)
Comment (CharlesDelannoy) [reply lib/safe-pg-migrations/plugins/idempotent_statements.rb:37]: I prefer to keep it that way because I prefer if instead of unless

**Review by MouadHilali — APPROVED** [review]

**Comment by CharlesDelannoy** [comment]
For the Readme I'll update it along with the version upgrade

File: Gemfile.lock [hunk 7]
```diff
@@ -2,27 +2,38 @@ PATH
   remote: .
   specs:
     safe-pg-migrations (2.3.1)
-      activerecord (>= 6.0, < 7.1.x)
-      activesupport (>= 6.0, < 7.1.x)
+      activerecord (>= 6.1)
+      activesupport (>= 6.1)
```
[the remark this answers is not in the document]
Comment (mjcbsn22) [reply Gemfile.lock:6]: This did not make it into the release notes 😞

**Pull request merged**